streams/src/main/java/org/apache/kafka/streams/kstream/internals/SessionWindowedKStreamImpl.java

```diff
@@ -289,7 +289,10 @@ private <VR> StoreBuilder<SessionStore<K, VR>> materialize(final MaterializedInt
         // do not enable cache if the emit final strategy is used
         if (materialized.cachingEnabled() && emitStrategy.type() != EmitStrategy.StrategyType.ON_WINDOW_CLOSE) {
             builder.withCachingEnabled();
+        } else {
+            builder.withCachingDisabled();
```

Comment:
Why do we need this call? I thought we only add tests in this PR? Is the feature not completed yet and we need this?

Reply:
Want to make it explicit in case default is caching enabled. I can remove it if you think it's unnecessary.

Reply:
Maybe @guozhangwang can chime in?

I am happy both ways I guess, but we should keep it consistent if possible.